src/CP/core/model.jl

```diff
@@ -212,6 +242,9 @@ function reset_model!(model::CPModel)
     model.objectiveBound = nothing
     empty!(model.statistics.solutions)
     empty!(model.statistics.nodevisitedpersolution)
+    for (key, value) in model.statistics.infeasibleStatusPerVariable
+        model.statistics.infeasibleStatusPerVariable[key]=length(getOnDomainChange(model.variables[key]))  #the degree is reset to the initial value : the number of constraints the variable is involved in. 
+    end   

```

Comment:
resetting the counter to the iniatial value : the lenght of the variable's domain.